biothings/hub/databuild/backend.py

```diff
@@ -158,7 +158,9 @@ def get_src_metadata(self):
             if src and src.get("download"):
                 # Store the latest success dump time
                 if src["download"]["status"] == "success":
-                    src_meta.setdefault(src["_id"], {}).setdefault("download_date", src["download"]["started_at"])
+                    src_meta.setdefault(src["_id"], {})
+                    src_meta[src["_id"]]["last_download_date"] = src["download"].get("last_time")
+                    src_meta[src["_id"]].setdefault("download_date", src["download"]["started_at"])
```

Comment:
@sengineer0 I actually did not mean to store "last_download_date" in src_meta (output in /metadata endpoint). Just need to store a "last _download_date" internally in mongodb collection, so that when we populate the correct "download_date" for src_meta if a source failed.

Reply:
https://github.com/newgene/biothings.api/pull/36/#issuecomment-1223505417
please check this comment :)